### DISCUSSION TITLE
Highlight the recommended difficulty

### DISCUSSION THREAD

**Discussion by Gabixel** [ppy/osu#13034] [Ideas]
**Describe the new feature:** The feature is simple: highlight the recommended difficulty of a player on the carousel.

The feature is already implemented on the _Beatmaps Listing_ web page with a filter, and I think that can be very cool to see it in-game too:
![image](https://user-images.githubusercontent.com/43073074/60278016-f4f01280-98fe-11e9-9412-5869f2345a8f.png)

**Proposal designs of the feature:** 

- The idea is to highlight the recommended difficulty, based on my idea, with a golden colour and a star on the left side (but that's optional):
![Recommended Difficulty](https://user-images.githubusercontent.com/43073074/60278234-57e1a980-98ff-11e9-8b70-14e0d8e6a680.png)

- I added that star for maybe a future proposal for other "warnings" (like missing maps with a `(!)` red icon), and the player can hover with the mouse to see what does it represent:
![Recommended Difficulty pt 2](https://user-images.githubusercontent.com/43073074/60278523-ea824880-98ff-11e9-8aeb-3f719591e1f4.png)

- And we can add another text saying something like "_This is the recommended difficulty_" on the info wedge (or even inside the "Details" tab):
![Recommended Difficulty pt 3](https://user-images.githubusercontent.com/43073074/60278885-73997f80-9900-11e9-9726-5e84c0c8ab92.png)
![Recommended Difficulty pt 4](https://user-images.githubusercontent.com/43073074/60279166-dc80f780-9900-11e9-8b4f-92405d3fceee.png)

**Pros:** A gold colour makes the recommended difficulty easy to see. If the player doesn't like that, he can disable the feature in the settings.
**Cons:** Many things are already coloured yellow/gold.. obviously I don't want to make everything yellow, so you can choose what's the best colour.

**Comment by nyquillerium** [comment]
dunno if this is something @arflyte would like to give input on

**Comment by LittleEndu** [comment]
Is the glow in [current design draft](https://www.figma.com/file/DXKwqZhD5yyb1igc3mKo1P/Song-Select-Draft-2?node-id=1%3A1276) selection glow or recommendation glow?

**Reply by frenzibyte** [reply]
Looks more like a selection glow to me, I think a recommendation indicator would be something along the lines of a golden tint to the beatmap card.

**Reply by smoogipoo** [reply]
@arflyte please provide input here. To me it looks like the selection is indicated by the background brightening, but not sure.

**Reply by peppy** [reply]
It's the selected item's glow, correct. Recommendation would be an icon, not a glow. It needs to be explicit - not something people have to guess at or look up.

**Reply by arflyte** [reply]
That's the selection glow. In the sample it glows yellow of the of the difficulty colour. I didn't put any thoughts on recommended difficulty in any of the designs. I can be something as simple as an icon, but I need to look at all the areas first before I can suggest something.

**Comment by Givikap120** [comment]
I would propose this kind of mechanism:

There are 3 values:
Upper bound = `player pp top play` * 1.2 + 10pp
Best match = arithmetic mean of `player pp top play` and `player profile total pp / 20` (to represent average top play)
Lower bound = `player top100 play`, 0pp if player have less than 100 scores

Then it checks through difficulties of the map and checks the amount of pp for FC with active mods and `recommended accuracy`. When there is at least one difficulty between lower and upper bound - it highlights the difficulty closest to best match.

`recommended accuracy` - return value of the function that shows expected accuracy based on the map's OD and player skill level represented in profile pp

**Reply by LittleEndu** [reply]
This is a discussion about the design, not about the recommendation algorithm (which should be a new discussion or https://github.com/ppy/osu-web/issues/9020)

**Reply by Givikap120** [reply]
this recommended difficulty (of the map) is based on pp, not star rating
converting pp to star rating is another issue
but I get what you meant to say

**Comment by smoogipoo** [comment]
To keep things simple, I don't think we should highlight the recommended difficulty at all. The game should do a reasonably good job of deciding for the user what they may want to play, without the user having to engage their brain with yet another informational resource.
